src/zenml/utils/source_utils.py

```diff
@@ -582,33 +590,61 @@ def _try_to_load_notebook_source(source: NotebookSource) -> Any:
 
     Raises:
         RuntimeError: If the source can't be loaded.
+        FileNotFoundError: If the file containing the notebook cell code can't
+            be found.
 
     Returns:
         The loaded object.
     """
-    if not source.code_path or not source.replacement_module:
+    if not source.replacement_module:
         raise RuntimeError(
             f"Failed to load {source.import_path}. This object was defined in "
             "a notebook and you're trying to load it outside of a notebook. "
-            "This is currently only enabled for ZenML steps."
+            "This is currently only enabled for ZenML steps and materializers. "
+            "To enable this for your custom classes or functions, use the "
+            "`zenml.utils.notebook_utils.enable_notebook_code_extraction` "
+            "decorator."
         )
 
     extract_dir = _get_shared_temp_dir()
-    file_path = os.path.join(extract_dir, f"{source.replacement_module}.py")
+    file_name = f"{source.replacement_module}.py"
+    file_path = os.path.join(extract_dir, file_name)
 
     if not os.path.exists(file_path):
+        from zenml.client import Client
         from zenml.utils import code_utils
 
+        artifact_store = Client().active_stack.artifact_store
+
+        if (
```

Comment:
If the notebook cell content is stored in a different artifact store than the active one we are failing here. Similar to what we did with the pipeline artifacts, could we not try to initialize this other artifact store and use it to load it?

Reply:
+1, seems like the old bug we had